### ISSUE TITLE
Middlewares are broken ?

### ISSUE THREAD

**Issue by nolway** [kartikk221/hyper-express#125]
Hello I tried to implement a simple middleware but the log is never displayed :
```ts
const webserver = new HyperExpress.Server();

// Global middlewares
const myLogger = function (req:Request, res:Response, next:MiddlewareNext) {
    console.log('LOGGED')
    next()
  }

webserver.use(myLogger);
```
Also I tried to do this in a route : 
```ts
const specific_middleware1 = (request:Request, response:Response, next:MiddlewareNext) => {
    console.log('route specific middleware 1 ran!');
    return next();
};
this.app.get("/map", { middlewares: [specific_middleware1]} , (req: Request, res: Response) => {
...
```
Error message :
```
TypeError: middleware.handler is not a function
    at Route.handle (/usr/src/app/node_modules/hyper-express/src/components/router/Route.js:95:37)
    at Server._handle_uws_request (/usr/src/app/node_modules/hyper-express/src/components/Server.js:391:19)
    at /usr/src/app/node_modules/hyper-express/src/components/Server.js:315:26
```

Did you have an idea ?
(HyperExpress in 6.4.10)

**Comment by kartikk221** [comment]
Hey @nolway I'm not able to re-create the behavior you are having. 

It seems in your first example, If you do not have any kind of a route assigned or a `not_found_handler` assigned then no middleware will run at all.

In your second example, I don't see what is wrong in this example, In fact there is a test written specifically for this use case here: https://github.com/kartikk221/hyper-express/blob/75fff1577ae2b90c7e9a025359299c3aefd170a4/tests/components/http/scenarios/middleware_double_iteration.js#L20

Would you be able to provide a mini app code to recreate this behavior? Thanks

**Comment by kartikk221** [comment]
I'll be closing this for now due to inactivity. Feel free to follow up once you find a way to recreate this behavior. Thanks

**Comment by moufmouf** [comment]
Found the issue we were having with @nolway !

We actually were calling:

```
server.uws_instance.listen(3000);
```

instead of:

```
server.listen(3000);
```

So the 'listen' function of hyper-express was never called. It worked like this for 1 year, but with the recent update of hyper-express, it started failing.

We solved the issue and everything is working fine now.

Thanks for your help @kartikk221 !

(by the way, we are using hyper-express extensively on WorkAdventure: https://github.com/thecodingmachine/workadventure, thanks a lot for hyper-express :+1: )
